tests/bugfixes/github/test_issue_20.py

```diff
@@ -26,6 +26,7 @@ class TamronSupport(metaclass=system_tests.CaseMeta):
         """Exif.CanonCs.LensType                        Short       1  Tamron SP 90mm f/2.8 Di VC USD Macro 1:1
 Exif.CanonCs.Lens                            Short       3  90.0 mm
 Exif.Canon.LensModel                         Ascii      70  TAMRON SP 90mm F/2.8 Di VC USD MACRO1:1 F017
+Exif.CanonLe.LensSerialNumber                SLong       1  0
```

Comment:
Serialnumber of `0` seems weird. Is this info not included in all files? 
Are there other tests where we can see if this decoding works?

After going through the entire PR, I don't see a single value that isn't zero... :thinking:

Reply:
@alexvanderberkel did that.  Alex?  That is a read/write field.  You can test it with ExifTool.  So, you can probably modify one of the existing test images to add a test for this.

```
734 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ exiv2 -K Exif.Image.Make *.jpg 2>/dev/null | grep Canon 
FurnaceCreekInn.jpg   Exif.Image.Make                              Ascii       6  Canon
FurnaceCreekInn2.jpg  Exif.Image.Make                              Ascii       6  Canon
FurnaceCreekInn3.jpg  Exif.Image.Make                              Ascii       6  Canon
exiv2-bug1202.jpg     Exif.Image.Make                              Ascii       6  Canon
exiv2-bug1231a.jpg    Exif.Image.Make                              Ascii       6  Canon
exiv2-bug1231b.jpg    Exif.Image.Make                              Ascii       6  Canon
exiv2-bug445.jpg      Exif.Image.Make                              Ascii       6  Canon
exiv2-bug501.jpg      Exif.Image.Make                              Ascii       6  Canon
exiv2-bug528.jpg      Exif.Image.Make                              Ascii       6  Canon
exiv2-bug876.jpg      Exif.Image.Make                              Ascii       6  Canon
exiv2-canon-eos-20d.jpg  Exif.Image.Make                              Ascii       6  Canon
exiv2-canon-eos-300d.jpg  Exif.Image.Make                              Ascii       6  Canon
exiv2-canon-eos-d30.jpg  Exif.Image.Make                              Ascii       6  Canon
exiv2-canon-powershot-a520.jpg  Exif.Image.Make                              Ascii       6  Canon
exiv2-canon-powershot-s40.jpg  Exif.Image.Make                              Ascii       6  Canon
issue_1912_poc.jpg    Exif.Image.Make                              Ascii       6  Canon
smiley2.jpg           Exif.Image.Make                              Ascii       6  Canon
735 rmills@rmillsm1:~/gnu/github/exiv2/main/test/data $ 
```

Reply:
will do that

Reply:
@hassec
```
879 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ exiftool CanonEOS_R6.jpg | grep -i serial
Internal Serial Number          : KQ0008148
Serial Number                   : 012021000024
Lens Serial Number              : 8003003431
880 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ exiv2 -K Exif.CanonLe.LensSerialNumber CanonEOS_R6.jpg 
Exif.CanonLe.LensSerialNumber                SLong       1  872416128
881 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ 
```

I've looked at the ExifTool test files and I believe this is working.  I documented how to obtain them: https://exiv2.org/book/index.html#8-6

**BUT BUT BUT _Wait a minute_ **
I extracted the .exv (with a view to putting it in the test suite) and it's not correct.

```
880 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ exiv2 -K Exif.CanonLe.LensSerialNumber CanonEOS_R6.jpg 
Exif.CanonLe.LensSerialNumber                SLong       1  872416128
881 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ exiv2 -ea --verbose CanonEOS_R6.jpg 
File 1/1: CanonEOS_R6.jpg
Writing Exif data from CanonEOS_R6.jpg to ./CanonEOS_R6.exv
Writing XMP data from CanonEOS_R6.jpg to ./CanonEOS_R6.exv
882 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ exiv2 -K Exif.CanonLe.LensSerialNumber CanonEOS_R6.exv
Exif.CanonLe.LensSerialNumber                SLong       1  28
883 rmills@rmillsmm-local:/Users/Shared/Jenkins/Home/userContent/testfiles/ExifTool/Canon $ 
```
If this PR is merged without solving this, we should open an issue concerning the .exv.

_**Other points**_

1.  We don't have a bug report for those canon tag changes. 
2.  We should investigate all the canon tag changes using the ExifTool test suite and compare our results with ExifTool.

Reply:
@hassec.  I was confused when I thought there was an issue concerning Exif.CanonLe.LensSerialNumber.  I have investigated and provided evidence that it is working correctly.  https://github.com/Exiv2/exiv2/pull/2087#issuecomment-1036322420